### PR TITLE
CP-566 Fluent URI library (fluri)

### PR DESCRIPTION
**Pull request by evanweible-wf**
# Fluri

> Fluri is a fluent URI library for Dart built to make URI mutation easy.
#### `Fluri`

> Use directly as a replacement for `Uri`.

``` dart
import 'package:fluri/fluri.dart';

void main() {
  Fluri uri = new Fluri()
    ..host = 'example.com'
    ..scheme = 'https'
    ..path = 'path/to/resource'
    ..queryParameters = {'limit': '10', 'format': 'json'};

  print(uri.toString());
  // https://example.com/path/to/resource?limit=10&format=json
}
```
#### `FluriMixin`

> Extend or use as a mixin to add the fluent URI mutation API to your own class.

``` dart
import 'package:fluri/fluri.dart';

// Option 1: Extending
class Request extends FluriMixin {}

// Option 2: Using as a Mixin
class Request extends Object with FluriMixin {}

void main() {
  Request req = new Request()
    ..host = 'example.com'
    ..scheme = 'https'
    ..path = 'path/to/resource'
    ..queryParameters = {'limit': '10', 'format': 'json'};

  print(req.uri.toString);
  // https://example.com/path/to/resource?limit=10&format=json
}
```
## API

See the code and inline docs for API reference. Can also pull this repo down and run `./tool/docs.sh`
## Tools

**Tests:** `./tool/test.sh`
**Documentation:** `./tool/docs.sh`
**Coverage:** `./tool/coverage.sh`

> To run coverage, you will need to install the `coverage` package globally:
> `pub global activate coverage` 
> And install lcov with brew:
> `brew install lcov`
## Test Coverage

screen shot 2015-04-23 at 2 28 38 pm

---

@trentgrover-wf
@maxwellpeterson-wf
@shanesizer-wf 
@johnryan-wf

### CODE REVIEW

**Comment by jayudey-wf** [comment]
go @rmconsole-wf

**Comment by jayudey-wf** [comment]
@rosie run_merge_script

**Comment by jayudey-wf** [comment]
@rmconsole-wf

**Comment by rmconsole-wf** [comment]
- [x] All commits reviewed
- [x] QA Review
- [x] Security Review Completed if Applicable:  (N/A)

To remain in SOC1 Compliance the following users cannot merge this PR:
* evanweible-wf
* maxwellpeterson-wf

		When this pull is merged I will use the following information:
		Version: fluri 1.0.0
		Release Ticket: CP-573

***
*Last updated on Tuesday, April 28 02:58 PM CST*

File: test/fluri_test.dart [hunk 1]
```diff
@@ -0,0 +1,107 @@
+library fluri.test.fluri_test;
+
+import 'package:fluri/fluri.dart';
+import 'package:test/test.dart';
+
+
+void commonFluriTests(dynamic getFluri()) {
```
Comment (maxwellpeterson-wf) [inline test/fluri_test.dart]: Purely curious - why does this take a function instead of the actual fluri object?

**Comment by teresarevious-wf** [comment]
@rmconsole-wf

**Pull request merged**